src/cow-react/modules/advancedOrders/hooks/useParseNumberOfParts.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from 'react'
+import { useSetAtom } from 'jotai'
+import {
+  updateAdvancedOrdersSettingsAtom,
+  NumberOfPartsError,
+} from '@cow/modules/advancedOrders/state/advancedOrdersSettingsAtom'
+import { MIN_PARTS_NUMBER, MAX_PARTS_NUMBER } from '@src/constants'
+
+export function useParseNumberOfParts() {
+  const updateSettingsState = useSetAtom(updateAdvancedOrdersSettingsAtom)
+
+  return useCallback(
+    (value: string) => {
+      const parsed = parseInt(value) || 0
+
+      updateSettingsState({ numberOfParts: parsed, numberOfPartsError: null })
+
+      if (parsed < MIN_PARTS_NUMBER || parsed > MAX_PARTS_NUMBER) {
+        updateSettingsState({ numberOfPartsError: NumberOfPartsError.InvalidInput })
+      }
```

Comment:
I think link 16 should be executed only if the validation passes:

```suggestion
      if (parsed < MIN_PARTS_NUMBER || parsed > MAX_PARTS_NUMBER) {
        updateSettingsState({ numberOfPartsError: NumberOfPartsError.InvalidInput })
      } else {
        updateSettingsState({ numberOfParts: parsed, numberOfPartsError: null })
      }
```

Reply:
I'm not sure that is the case because we want to let user to be able to insert any value, and then we will show him the error if the value is not acceptable.

Reply:
Ah I see. The goal of the early update is to reset "bad" user input.
Then all good